baseplate/experiments/__init__.py

```diff
@@ -134,25 +138,43 @@ def variant(self, name, user=None, bucketing_event_override=None,
         :rtype: :py:class:`str`
         :return: Variant name if a variant is active, None otherwise.
         """
+
         experiment = self._get_experiment(name)
 
         if experiment is None:
             return None
 
         inputs = dict(kwargs)
+
+        """ clean up user id. If not a baseplate user, we risk inconsistency."""
         if user:
             if user.is_logged_in:
                 inputs["user_id"] = user.id
                 inputs["user_roles"] = user.roles
+                inputs["logged_in"] = user.is_logged_in
             else:
                 inputs["user_id"] = user.loid
-            inputs["logged_in"] = user.is_logged_in
+                inputs["logged_in"] = False
+
+        #loid is numeric. Convert it to a base-36 string; Ideally, this will
```

Comment:
`user.loid` should already be in the fullname format: https://github.com/reddit/baseplate/blob/master/baseplate/core.py#L419

Reply:
only if it's passed in via a baseplate user object. the loid as it exists in the r2 context is numeric, and if someone passes it in via `**kwargs`, I want to make sure we've got it formatted correctly.

Reply:
I don't really think that we want to try and fix that in baseplate though, the framework should just assume that the input you passed in is what you want to use.  I think if we tried to "fix" this, we would actually rebucket every logged out experiment since r2 always used the full `loid.id` value, not the fullname format, for bucketing.

Reply:
Agreed. Baseplate's style is usually to yell at you loudly if you do something wrong rather than silently fix it. Since this would be a problem only in r2 (and only until we get the stuff wired up like a baseplate app there) we can just put any value-massaging code in r2 and leave baseplate strict and sassy.